tests/sentry/uptime/consumers/test_results_consumers.py

```diff
@@ -119,3 +129,138 @@ def test_missed(self):
         hashed_fingerprint = md5(str(self.project_subscription.id).encode("utf-8")).hexdigest()
         with pytest.raises(Group.DoesNotExist):
             Group.objects.get(grouphash__hash=hashed_fingerprint)
+
+    def test_onboarding_failure(self):
+        self.project_subscription.update(
+            mode=ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING
+        )
+        result = self.create_uptime_result(
+            self.subscription.subscription_id,
+            status=CHECKSTATUS_FAILURE,
+            scheduled_check_time=datetime.now() - timedelta(minutes=5),
+        )
+        redis = _get_cluster()
+        key = build_onboarding_failure_key(self.project_subscription)
+        assert redis.get(key) is None
+        with mock.patch("sentry.uptime.consumers.results_consumer.metrics") as metrics:
+            self.send_result(result)
+            metrics.incr.assert_has_calls(
+                [
+                    call(
+                        "uptime.result_processor.handle_result_for_project",
+                        tags={"status": CHECKSTATUS_FAILURE, "mode": "auto_detected_onboarding"},
+                    ),
+                ]
+            )
+        assert redis.get(key) == "1"
+
+        hashed_fingerprint = md5(str(self.project_subscription.id).encode("utf-8")).hexdigest()
```

Comment:
I wish there was a function that did this lol. Can't use `md5_text` here?

Reply:
Yeah maybe I should make a function for this since we use it a bunch, or build the check into some framework at some point.

Reply:
Can't change this unfortunately since it needs to match the method we use in issue platform